R/addins.R

```diff
@@ -105,6 +112,31 @@ style_selection <- function() {
   }
 }
 
+style_selection_v2 <- function() {
+  communicate_addins_style_transformers()
+  id <- rstudioapi::documentId()
+  selection <- rstudioapi::selectionGet(id = id)
+  text <- selection$value
+  if (!nzchar(text)) abort("No code selected")
+  out <- style_text(text, transformers = get_addins_style_transformer())
+  rstudioapi::selectionSet(value = out, id = id)
+
+  if (Sys.getenv("save_after_styling") == TRUE &&
+      !is.null(rstudioapi::documentPath(id))) {
+    invisible(rstudioapi::documentSave(id))
+  }
+
+}
+
+rstudio_has_selection_apis <- function() {
+
+  rstudioapi::hasFun("selectionGet") &&
```

Comment:
I am afraid this is a bit slow (given that styler is already quite slow). Is making it RStudio version dependent an option?

Reply:
Yes, that would also be suitable (`rstudioapi >= 0.12`; `RStudio >= 1.4.845`). Or, this condition could be cached and reused.